generator/src/main/java/com/graphicsfuzz/generator/knownvaluegeneration/ExpressionGenerator.java

```diff
@@ -561,7 +571,7 @@ private String parseNameFromValue(Value value) {
     if (value instanceof NumericValue) {
       final NumericValue numericValue = (NumericValue) value;
       float floatValue = numericValue.getValue().get().floatValue();
-      if (floatValue < 0.0) {
+      if (String.valueOf(floatValue).contains("-")) {
```

Comment:
I found that we could end up having a random float -0.0 making the previous version of this condition does not work. Thus, I believe that to know whether the number is negative, checking if the string contains "-" would be nicer.